### PR TITLE
Qualifier values do not have first character converted to lowercase

### PR DESCRIPTION
**Pull request by wetterjames4**
FromString panics for some inputs. This change fixes the issue and adds a test to ensure it works.

### CODE REVIEW

File: packageurl.go [hunk 1]
```diff
@@ -482,11 +482,14 @@ func parseQualifiers(rawQuery string) (Qualifiers, error) {
 			return nil, fmt.Errorf("error unescaping qualifier value %q", value)
 		}
 
-		q = append(q, Qualifier{
-			Key: strings.ToLower(key),
+		if len(value) > 0 {
 			// only the first character needs  to be lowercase. Note that pURL is always UTF8, so we
 			// don't need to care about unicode here.
-			Value: strings.ToLower(value[:1]) + value[1:],
+			value = strings.ToLower(value[:1]) + value[1:]
```
Comment (wetterjames4) [inline packageurl.go:488]: I am not even sure the lowercase first character is needed. I can't see it specified in the spec.

Comment (shibumi) [reply packageurl.go:488]: @tommyknows  Hey Ramon, do you know if we really need lowercase characters here?

Comment (tommyknows) [reply packageurl.go:488]: Mh, I thought this was just the behaviour I ported from before the refactor, but I actually can't seem to find that 👀 

i wonder why I've added it in that case... 

Also, it's not in the spec and I can't see any test cases requiring this. I'd say remove it then...

Comment (wetterjames4) [reply packageurl.go:483]: I am in favour of removing the lowercase conversion. I updated the title of the pull request and added another commit to just remove the case conversion.

**Comment by shibumi** [comment]
I think we can close this PR here, because the commits are already in: https://github.com/package-url/packageurl-go/pull/65

**Pull request closed without merging**